domain/schema/model/sql/0022-resource.sql

```diff
@@ -0,0 +1,108 @@
+CREATE TABLE resource_origin_kind (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_origin_name
+ON resource_origin_kind (name);
+
+INSERT INTO resource_origin_kind VALUES
+(0, 'uploaded'),
+(1, 'store');
+
+CREATE TABLE resource_state (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_state
+ON resource_state (name);
+
+INSERT INTO resource_state VALUES
+(0, 'available'),
+(1, 'potential');
+
+CREATE TABLE resource (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    application_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    origin_kind_id INT NOT NULL,
+    state_id INT NOT NULL,
+    size INT,
+    fingerprint TEXT,
```

Comment:
Can we rename this to hash (we can leave the code as fingerprint) and then have a hash_type_id similar to the object store. This will then allow us to migrate away from the old hash to a new one when required.

https://github.com/juju/juju/blob/279abbe7c068c9f6dc7683085ecf986b80ea129e/domain/schema/model/sql/0006-objectstore-metadata.sql#L15-L16


https://github.com/juju/juju/blob/279abbe7c068c9f6dc7683085ecf986b80ea129e/domain/schema/model/sql/0006-objectstore-metadata.sql#L1-L11

Reply:
done